### PR TITLE
[DOC][NFC] Batchnorm inference API documentation update

### PR DESCRIPTION
**Pull request by muralinr**
Batchnorm inference API documentation is updated with more details

### CODE REVIEW

**Review by junliume — APPROVED** [review]

**Comment by carlushuang** [comment]
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/batch_norm_api_doc_update/1/pipeline

this PR not passed CI, now it becomes a blocker for our CI process. #1452

**Comment by junliume** [comment]
> http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/batch_norm_api_doc_update/1/pipeline
> 
> this PR not passed CI, now it becomes a blocker for our CI process. #1452

Thank you for the hotfix!
I skipped CI since it is doc-only change, did not know that clang will check comments' format too. Will be more careful next time.
CC: @atamazov  @JehandadKhan

**Comment by atamazov** [comment]
@junliume Yes. We can run NOGPU-only tests on CI to quickly check formatting (and, IIRC doxygen-related problems, which are quite possible when documentation is changed).
